pkg/cluster/node.go

```diff
@@ -297,7 +297,13 @@ func patchServerSpec(node *k3d.Node) error {
 	node.Labels[k3d.LabelServerAPIHost] = node.ServerOpts.ExposeAPI.Host
 	node.Labels[k3d.LabelServerAPIPort] = node.ServerOpts.ExposeAPI.Port
 
-	node.Args = append(node.Args, "--tls-san", node.ServerOpts.ExposeAPI.Host) // add TLS SAN for non default host name
+	// If the runtime is docker, attempt to use the docker host
+	if runtime.ID() == "docker" && runtime.GetHost() != "" {
+		node.Labels[k3d.LabelServerAPIHostIP] = runtime.GetHost()
+		node.Labels[k3d.LabelServerAPIHost] = runtime.GetHost()
+	}
+
+	node.Args = append(node.Args, "--tls-san", node.Labels[k3d.LabelServerAPIHost]) // add TLS SAN for non default host name
```

Comment:
Shouldn't we then already error out if a different Host was set for the ExposeAPI.Host?

Reply:
I don't believe so, but this is also where things get a bit wonky. From what I could tell the APIHost and APIHostIP almost exclusively assumed local host and 0.0.0.0 was going to be the value. 

I originally tried just overriding APIHostIP globally at run-time via a `--host` parameter but that severely broke things in a number of ways, to include docker port forwarding. 

That's when I found the TODO note a few lines up stating maybe this should be the IP of the docker host, that's when I added the runtime.ID() and runtime.GetHost() so that if we are on a remote host we set it on the label.

Reply:
Yep, what I meant was that we should throw an error or a warning when a user uses a remote docker connection (i.e. has the `DOCKER_HOST` env var set) but also tries to override the APIHost in the `--api-port` flag.

Reply:
I see, yes that could be a problem.